modules/logrotate/templates/logrotate.conf.erb

```diff
@@ -43,6 +43,16 @@
     # Remove empty .1 log files. There seems to be an issue where logrotate
     # will refuse to rotate the logs if an empty .1 log file exists.
     find <%= @matches %>* -wholename '/var/log/*' -name '*.1' -size 0 -delete
+    # Remove files bigger than $SIZE in chronological order while use of  
+    # disk space on the logging partition is greater than $MAX_USE
+    SIZE='300M'
+    MAX_USE=85
+    for file in $(find <%= @matches %>* -type f -size +$SIZE -exec ls -tr {} +); do
+      USE=$(df /var/log --output=pcent | sed -n '2p' | tr -d %)
+        if [ $USE -gt $MAX_USE ]; then
+          rm $file
```

Comment:
If I understand this correctly, we loop over every log file (that is more than 300M), and check if disk space is at >85%. If it is, remove the file. Once we reach <=85%, we won't be removing any more, but we'll still be looping over the rest of the files.

I think it would be logically easier to follow by inverting it like this:

```
while (not enough disk space)
  remove the oldest file that is >300mb
```

But this is quite nitpicky 😅 also not sure how slow the `find` query is, in which case my suggestion would be less efficient. So happy to leave as-is 👍

Reply:
Yes that's the idea, and I agree the logic of the loop is a bit backward.
The reason for that is mainly to keep it simple as find gives us a nice list to go over in a simple way, using the disk space left to control the loop would make more sense but would introduce more control work as we would need some bit of logic to break the loop if we run out of files to delete.
As it is this we just get a list of files we think we can do something about, work and them, and then we've done what we could to alleviate the space issue.